source/apiVolontaria/apiVolontaria/views.py

```diff
@@ -115,6 +116,52 @@ def get_authenticators(self):
             self.authentication_classes = ()
         return [auth() for auth in self.authentication_classes]
 
+    def post(self, request, *args, **kwargs):
+        response = self.create(request, *args, **kwargs)
+        # get token for user save and send key this token with a mail
+        user = User.objects.get(username=request.data["username"])
+
+        if response.status_code == status.HTTP_201_CREATED:
+            if settings.CONSTANT['AUTO_ACTIVATE_USER'] is True:
+                user.is_active = True
+                user.save()
+
+            if settings.CONSTANT['EMAIL_SERVICE'] is True:
```

Comment:
I think this `if` condition may be a `elif` condition. Actually you can activate the account AND send an email to activate, it can be strange for the user

Reply:
> This is for documentation purpose, we agreed to keep it this way for now

I see it this way:

- If `AUTO_ACTIVATE_USER = True` and `EMAIL_SERVICE = True`
The email sent corresponds to a "confirmation" email. That is just a feedback from the website and no action needs to be taken by the user to activate his account.

- If `AUTO_ACTIVATE_USER = False` and `EMAIL_SERVICE = True`
The email sent corresponds to an "activation" email. That is a feedback from the website but the user also need to take action to activate his account.

- If `AUTO_ACTIVATE_USER = True` and `EMAIL_SERVICE = False`
User is activated automatically and no email is sent.

- If `AUTO_ACTIVATE_USER = False` and `EMAIL_SERVICE = False`
No email is sent and user is not activated automatically. This situation requires an administrator to activate user accounts manually.